Apply changes of various PRs

tarteo opened the pull request.

carlosdauden committed:
[9.0][ADD] contract_mandate: New module

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[MIG] contract_mandate: Migration to 10.0

carlosdauden committed:
[IMP] contract_variable_qty_timesheet: New module

carlosdauden committed:
[ADD] contract/contract_sale_invoicing: New module

sbejaoui committed:
[MIG] contract_sale_invoicing: Migration to 12.0

sbejaoui committed:
[IMP] - get contract line default data onchange product_id

ThomasBinsfeld committed:
[MIG] Contract Mandate to 12.0

primes2h committed:
Added translation using Weblate (Italian)

Currently translated at 66.7% (2 of 3 strings)

Translation: contract-11.0/contract-11.0-contract_variable_qty_timesheet
Translate-URL: https://translation.odoo-community.org/projects/contract-11-0/contract-11-0-contract_variable_qty_timesheet/it/

marylla committed:
Added translation using Weblate (German)

Currently translated at 100.0% (3 of 3 strings)

Translation: contract-11.0/contract-11.0-contract_variable_qty_timesheet
Translate-URL: https://translation.odoo-community.org/projects/contract-11-0/contract-11-0-contract_variable_qty_timesheet/de/

sbejaoui committed:
[IMP] - improve sale order line view

sbejaoui committed:
[FIX] - Fix unit tests

cubells committed:
[ADD] contract_price_revision: new module to update contract prices easily

pedrobaeza committed:
[IMP] contract_price_revision: Improve a bit performance

oca-travis committed:
[UPD] Update contract_price_revision.pot

ThomasBinsfeld committed:
[REF] Contract Sale Invoicing: split from analytic account

[REF] Contract Sale Invoicing: update translations

[IMP] - Assert that the predecessor is available for new link at uncancel

[RMV] - remove usless changes

[RMV] - Remove usless field recurring_invoices

after the total isolation between contract model and account analytic one.
recurring_invoices which was used to mark analytic account as contract became usless

[IMP] - P3 syntax

[IMP] - use @openupgrade.migrate() and openupgrade.logged_query

[IMP] - drop transient table in migration script

sbejaoui committed:
[12.0][REF] - Product Contract Variable quantity: split from analytic account

sbejaoui committed:
fix pylint

sbejaoui committed:
[FIX] - Fix unit tests

ThomasBinsfeld committed:
[REF] Contract Mandate: remove unused objects in unit tests

sbejaoui committed:
[12.0][FIX] - Fix unit tests

sbejaoui committed:
[IMP] - Improve unit tests

[UPD] Update product_contract_variable_quantity.pot

[UPD] README.rst

marylla committed:
Translated using Weblate (German)

Currently translated at 50.0% (3 of 6 strings)

Translation: contract-12.0/contract-12.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract_mandate/de/

britoederr committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (6 of 6 strings)

Translation: contract-12.0/contract-12.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract_mandate/pt_BR/

gurneyalex committed:
[FIX] add missing newlines in readme files

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (6 of 6 strings)

Translation: contract-12.0/contract-12.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract_mandate/zh_CN/

Martronic-SA committed:
contract_sale_invoicing fix moved field

Since that commit : https://github.com/OCA/contract/blob/a429b10c20cfd7b40064cbfdfd0313a5b1e27007, account_analytic_id has been moved to group_id.

Martronic-SA committed:
Update test account_analytic_id -> group_id

replaced contract's account_analytic_id by group_id

sbejaoui committed:
[12.0][FIX] - contract_mandate: fix inherited method call _prepare_invoice

sbejaoui committed:
Update contract_mandate/models/contract.py

Co-Authored-By: Pedro M. Baeza <pedro.baeza@tecnativa.com>

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100.0% (6 of 6 strings)

Translation: contract-12.0/contract-12.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract_mandate/de/

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: contract-12.0/contract-12.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract_mandate/es/

pedrobaeza committed:
[MIG] contract_variable_qty_timesheet: Migration to 12.0

* Standard procedure
* README by fragments
* Promoted to Production/Stable and declared maintainers
* Code adapted to v12
* Tests added

carlosdauden committed:
[MIG] contract_price_revision: Migration to 12.0

Commit:
Added translation using Weblate (Italian)

Commit:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: contract-12.0/contract-12.0-product_contract_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-product_contract_variable_quantity/it/

marcelsavegnago committed:
Added translation using Weblate (Portuguese (Brazil))

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 66.7% (2 of 3 strings)

Translation: contract-12.0/contract-12.0-contract_variable_qty_timesheet
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract_variable_qty_timesheet/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (7 of 7 strings)

Translation: contract-12.0/contract-12.0-product_contract_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-product_contract_variable_quantity/pt_BR/

astirpe committed:
[FIX] contract_price_revision: Skip revising lines that are not valid anymore
cherry-pick db5c3d647225dde2b20214932a060aa279554b07

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (7 of 7 strings)

Translation: contract-12.0/contract-12.0-product_contract_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-product_contract_variable_quantity/pt/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 16.7% (1 of 6 strings)

Translation: contract-12.0/contract-12.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract_mandate/pt/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (6 of 6 strings)

Translation: contract-12.0/contract-12.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract_mandate/pt/

Commit:
[FIX] product_contract_variable_quantity: use property_contract_template_id

[UPD] README.rst

[UPD] README.rst

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (7 of 7 strings)

Translation: contract-12.0/contract-12.0-product_contract_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-product_contract_variable_quantity/pt/

javierjcf committed:
[FIX]contract_mandate: Search mandate from correct company if we get in this method by the contract cron

fcolus committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (6 of 6 strings)

Translation: contract-12.0/contract-12.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract_mandate/pt_BR/

pdyharshad committed:
[FIX] looping contract in recurring_create_invoice to support multi recordset

fcolus committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (7 of 7 strings)

Translation: contract-12.0/contract-12.0-product_contract_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-product_contract_variable_quantity/pt_BR/

sbejaoui committed:
[ADD] - add contract_queue_job and contract_invoice_auto_validate

Commit:
[IMP] contract_mandate: black, isort, prettier

Commit:
[MIG] contract_mandate: Migration to 13.0

sbejaoui committed:
[FIX] - remvoe unnecessary utf8 coding comment

sbejaoui committed:
[FIX] - fix development_status

OCA-git-bot committed:
product_contract_variable_quantity 12.0.1.0.1

Commit:
[IMP] contract_variable_qty_timesheet: black, isort

Commit:
[MIG][13.0] contract_variable_qty_timesheet: Migration to 13.0

remi-filament committed:
Added translation using Weblate (French)

remi-filament committed:
Translated using Weblate (French)

Currently translated at 100.0% (7 of 7 strings)

Translation: contract-12.0/contract-12.0-product_contract_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-product_contract_variable_quantity/fr/

rousseldenis committed:
[IMP] contract_price_revision: black, isort, prettier

rousseldenis committed:
[13.0][MIG] contract_price_revision

rousseldenis committed:
[13.0][IMP] contract_price_revision: Add fixed price revision

rousseldenis committed:
[13.0][FIX] contract_price_revision: Recurring next date is not well computed

On new line creation when applying wizard, the stored values
are not recomputed, so recomputing them

rousseldenis committed:
[13.0][FIX] contract_price_revision: Copy() does the job

Commit:
[IMP] contract_queue_job: jobify renew method for crons

hkapatel-initos committed:
[MIG] contract_mandate: Migration to 14.0

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (9 of 9 strings)

Translation: contract-14.0/contract-14.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-14-0/contract-14-0-contract_mandate/es_AR/

carlosdauden committed:
[ADD] agreement_rebate_partner_company_group: New module that apply agreement to all company group members
TT34815

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (6 of 6 strings)

Translation: contract-14.0/contract-14.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-14-0/contract-14-0-contract_mandate/ca/

scigghia committed:
Translated using Weblate (Italian)

Currently translated at 50.0% (3 of 6 strings)

Translation: contract-14.0/contract-14.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-14-0/contract-14-0-contract_mandate/it/

alexis-via committed:
[FIX] Add check_company=True on contracts and contract lines

Add check_company=True in field definition and
[('company_id', '=', company_id)] in the domain (in view or in field definition)

rousseldenis committed:
[13.0][ADD] copier

rousseldenis committed:
[FIX] contract_price_revision: Use digits string instead of decimal precision

rousseldenis committed:
[IMP] contract_price_revision: Improve readability by adding a computed field 'price_can_be_revised'

rousseldenis committed:
[IMP] contract_price_revision: Improve compatibility with contract base module

Values for new line should come from _prepare_value_for_plan_successor function
adding the new price.

rousseldenis committed:
[IMP] contract_price_revision: Allows to restrict price revision on some lines

As if recurrence management is per line, some lines should be excluded from price revision.

BurkhalterY committed:
[IMP] contract_queue_job: black, isort, prettier

BurkhalterY committed:
[MIG] contract_queue_job: Migration to 14.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: contract-12.0/contract-12.0-product_contract_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-product_contract_variable_quantity/it/

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-product_contract_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-product_contract_variable_quantity/

rousseldenis committed:
[FIX] Development statuses

Commit:
[IMP] agreement_rebate_partner_company_group: black, isort, prettier

Commit:
[MIG] agreement_rebate_partner_company_group: Migration to 15.0

[UPD] Update agreement_rebate_partner_company_group.pot

[UPD] README.rst

Commit:
[UPD] Update contract_queue_job.pot

OCA-git-bot committed:
[UPD] README.rst

stefan-tecnativa committed:
[MIG] agreement_rebate_partner_company_group: Migration to 16.0

[UPD] Update agreement_rebate_partner_company_group.pot

OCA-git-bot committed:
[UPD] README.rst

carlos-domatix committed:
[15.0][ADD] subscription_oca

ilyasProgrammer committed:
[IMP] subscription_oca: tests

ilyasProgrammer committed:
[IMP] subscription_oca: refactoring

carolinafernandez-tecnativa committed:
[IMP] contract_price_revision: pre-commit stuff

carolinafernandez-tecnativa committed:
[MIG] contract_price_revision: Migration to 15.0

carolinafernandez-tecnativa committed:
[MIG] contract_variable_qty_timesheet: Migration to 15.0

[UPD] Update contract_variable_qty_timesheet.pot
[BOT] post-merge updates
Update translation files
Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-15.0/contract-15.0-contract_variable_qty_timesheet
Translate-URL: https://translation.odoo-community.org/projects/contract-15-0/contract-15-0-contract_variable_qty_timesheet/

carolinafernandez-tecnativa committed:
[IMP] contract_sale_invoicing: pre-commit stuff

ilyasProgrammer committed:
[IMP] subscription_oca: tests

carolinafernandez-tecnativa committed:
[MIG] contract_sale_invoicing: Migration to 15.0

TT45293

Co-Authored-By: Pedro M. Baeza <pedro.baeza@tecnativa.com>

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (23 of 23 strings)

Translation: contract-15.0/contract-15.0-contract_price_revision
Translate-URL: https://translation.odoo-community.org/projects/contract-15-0/contract-15-0-contract_price_revision/es/

Ivorra78 committed:
Added translation using Weblate (Spanish)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: contract-12.0/contract-12.0-product_contract_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-product_contract_variable_quantity/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (9 of 9 strings)

Translation: contract-14.0/contract-14.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-14-0/contract-14-0-contract_mandate/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: contract-14.0/contract-14.0-contract_queue_job
Translate-URL: https://translation.odoo-community.org/projects/contract-14-0/contract-14-0-contract_queue_job/es/

carolina-domatix committed:
[MIG] subscription_oca: Migration to 16.0

carolina-domatix committed:
[IMP] subscription_oca: tests adjustments

Commit:
[UPD] Update subscription_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

gregory-moka committed:
Added translation using Weblate (French)

ThijsvOers committed:
Added translation using Weblate (Dutch)

Commit:
[IMP] contract_mandate: pre-commit stuff

ThijsvOers committed:
Translated using Weblate (Dutch)

Currently translated at 95.7% (159 of 166 strings)

Translation: contract-16.0/contract-16.0-subscription_oca
Translate-URL: https://translation.odoo-community.org/projects/contract-16-0/contract-16-0-subscription_oca/nl/

luisDIXMIT committed:
[MIG] contract_queue_job: Migration to 16.0

Commit:
[UPD] Update contract_queue_job.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-16.0/contract-16.0-contract_queue_job
Translate-URL: https://translation.odoo-community.org/projects/contract-16-0/contract-16-0-contract_queue_job/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: contract-16.0/contract-16.0-agreement_rebate_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/contract-16-0/contract-16-0-agreement_rebate_partner_company_group/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 75.0% (3 of 4 strings)

Translation: contract-16.0/contract-16.0-contract_queue_job
Translate-URL: https://translation.odoo-community.org/projects/contract-16-0/contract-16-0-contract_queue_job/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: contract-16.0/contract-16.0-contract_queue_job
Translate-URL: https://translation.odoo-community.org/projects/contract-16-0/contract-16-0-contract_queue_job/it/

Commit:
[MIG][16.0] contract_mandate

Tisho99 committed:
[IMP] subscription_oca: pre-commit auto fixes

Tisho99 committed:
[MIG] subscription_oca: Migration to 17.0

Commit:
[IMP] contract_mandate: Increase test coverage

[UPD] Update contract_mandate.pot

Commit:
[UPD] Update subscription_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-17.0/contract-17.0-subscription_oca
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-subscription_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: contract-16.0/contract-16.0-contract_queue_job
Translate-URL: https://translation.odoo-community.org/projects/contract-16-0/contract-16-0-contract_queue_job/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (163 of 163 strings)

Translation: contract-17.0/contract-17.0-subscription_oca
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-subscription_oca/it/

juanjosesegui-tecnativa committed:
[IMP] contract_price_revision: pre-commit auto fixes

juanjosesegui-tecnativa committed:
[MIG] contract_price_revision: Migration to 17.0

[UPD] Update contract_price_revision.pot

[BOT] post-merge updates

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-17.0/contract-17.0-contract_price_revision
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-contract_price_revision/

juanjosesegui-tecnativa committed:
[IMP] contract_sale_invoicing: pre-commit auto fixes

juanjosesegui-tecnativa committed:
[MIG] contract_sale_invoicing: Migration to 17.0

danypr92 committed:
[MIG] contract_variable_qty_timesheet: Migration to 16.0

jarmokortetjarvi committed:
Translated using Weblate (Finnish)

Currently translated at 92.6% (151 of 163 strings)

Translation: contract-17.0/contract-17.0-subscription_oca
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-subscription_oca/fi/

Added translation using Weblate (Finnish)

metingulsoy committed:
Translated using Weblate (Turkish)

Currently translated at 61.3% (100 of 163 strings)

Translation: contract-17.0/contract-17.0-subscription_oca
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-subscription_oca/tr/

Translated using Weblate (Turkish)

Currently translated at 58.8% (96 of 163 strings)

Translation: contract-17.0/contract-17.0-subscription_oca
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-subscription_oca/tr/

Added translation using Weblate (Turkish)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: contract-16.0/contract-16.0-contract_mandate
Translate-URL: https://translation.odoo-community.org/projects/contract-16-0/contract-16-0-contract_mandate/it/

juanjosesegui-tecnativa committed:
[IMP] contract_variable_qty_timesheet: pre-commit auto fixes

juanjosesegui-tecnativa committed:
[MIG] contract_variable_qty_timesheet: Migration to 17.0

TT50112

Commit:
[UPD] Update contract_sale_invoicing.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update subscription_oca.pot

Commit:
[UPD] Update contract_variable_qty_timesheet.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-17.0/contract-17.0-subscription_oca
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-subscription_oca/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-17.0/contract-17.0-contract_sale_invoicing
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-contract_sale_invoicing/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: contract-17.0/contract-17.0-contract_sale_invoicing
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-contract_sale_invoicing/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: contract-17.0/contract-17.0-contract_variable_qty_timesheet
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-contract_variable_qty_timesheet/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (164 of 164 strings)

Translation: contract-17.0/contract-17.0-subscription_oca
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-subscription_oca/it/

mymage committed:
Added translation using Weblate (Italian)

Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: contract-17.0/contract-17.0-contract_price_revision
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-contract_price_revision/it/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (164 of 164 strings)

Translation: contract-17.0/contract-17.0-subscription_oca
Translate-URL: https://translation.odoo-community.org/projects/contract-17-0/contract-17-0-subscription_oca/tr/

victoralmau committed:
[ADD] contract_analytic_tag: New module

TT51221

[UPD] Update contract_analytic_tag.pot

[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: contract-16.0/contract-16.0-contract_analytic_tag
Translate-URL: https://translation.odoo-community.org/projects/contract-16-0/contract-16-0-contract_analytic_tag/it/

sbejaoui committed:
[18.0][MIG] product_contract_variable_quantity

BhaveshHeliconia committed:
[IMP] contract_mandate: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] contract_mandate: Migration to 17.0

tobiaszehntner committed:
[IMP] contract_sale_invoicing: pre-commit auto fixes

tobiaszehntner committed:
[MIG] contract_sale_invoicing: Migration to 18.0

Sale Order on 18.0 has no longer the field `analytic_account_id`. Instead, a variable distribution of multiple Analytic Accounts per Sale Order Line is possible. We adapted the module to consider only Sale Order Lines that match with 100%* on the analytic account of the contract.

AlvaroRM11 committed:
[MIG] contract_price_revision: Migration to 18.0

[FIX] set active_ids to list

[FIX] contract_price_revision: precommit run

[FIX] contract_price_revision: Change dependency

Co-authored-by: Alvaro <alvaramar@gmail.com>

TT57315

tobiaszehntner committed:
[MIG] contract_variable_qty_timesheet: Migration to 18.0

tobiaszehntner committed:
[IMP] subscription_oca: pre-commit auto fixes

tobiaszehntner committed:
[MIG] subscription_oca: Migration to 18.0

[IMP] subscription_oca: Use read_group for subscription count

Apply changes of https://github.com/OCA/contract/pull/1058

Apply changes of https://github.com/OCA/contract/pull/1224

Apply changes of https://github.com/OCA/contract/pull/1210

Apply changes of https://github.com/OCA/contract/pull/1167

subscription_oca: use get_formview_action

Fix close_reason_id visibility

Apply suggestions from review

Add "odoo 18"-like icon

sbejaoui committed:
[18.0][MIG] contract_queue_job

sbejaoui committed:
[18.0][MIG] contract_mandate

edescalona committed:
[IMP] contract_queue_job: pre-commit auto fixes

edescalona committed:
[MIG] contract_queue_job: Migration to 17.0

[17.0][IMP] contract_queue_job: Update manifest

acsonefho commented on subscription_oca/data/sale_subscription_data.xml:
```diff
@@ -11,61 +11,63 @@
 
     <data noupdate="1">
         <record id="subscription_stage_draft" model="sale.subscription.stage">
-            <!-- <field eval="True" name="active"/> -->
```
Please remove if useless

acsonefho commented on subscription_oca/data/sale_subscription_data.xml:
```diff
@@ -11,61 +11,63 @@
         </record>
 
         <record id="subscription_stage_in_progress" model="sale.subscription.stage">
-            <!-- <field eval="True" name="active"/> -->
```
Same here

acsonefho commented on subscription_oca/data/sale_subscription_data.xml:
```diff
@@ -11,61 +11,63 @@
         </record>
 
         <record id="subscription_stage_closed" model="sale.subscription.stage">
-            <!-- <field eval="True" name="active"/> -->
```
Same here

acsonefho commented on subscription_oca/data/sale_subscription_data.xml:
```diff
@@ -11,61 +11,63 @@
     </data>
 
     <record id="close_reason_expensive" model="sale.subscription.close.reason">
-        <!-- <field eval="True" name="active"/> -->
```
Same here

acsonefho commented on subscription_oca/data/sale_subscription_data.xml:
```diff
@@ -11,61 +11,63 @@
     </record>
 
     <record id="close_reason_requirement" model="sale.subscription.close.reason">
-        <!-- <field eval="True" name="active"/> -->
```
Same here

acsonefho commented on subscription_oca/data/sale_subscription_data.xml:
```diff
@@ -11,61 +11,63 @@
     </record>
 
     <record id="close_reason_ended" model="sale.subscription.close.reason">
-        <!-- <field eval="True" name="active"/> -->
```
Same here

acsonefho commented on subscription_oca/data/sale_subscription_data.xml:
```diff
@@ -11,61 +11,63 @@
     </record>
 
     <record id="close_reason_use" model="sale.subscription.close.reason">
-        <!-- <field eval="True" name="active"/> -->
```
Same here

acsonefho commented on subscription_oca/data/sale_subscription_data.xml:
```diff
@@ -11,61 +11,63 @@
     </record>
 
     <record id="close_reason_other" model="sale.subscription.close.reason">
-        <!-- <field eval="True" name="active"/> -->
```
Same here

acsonefho commented on subscription_oca/models/account_move.py:
```diff
@@ -10,3 +10,11 @@ class AccountMove(models.Model):
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "subscription_oca.sale_subscription_action"
+        )
+        action["domain"] = [("id", "=", self.subscription_id.id)]
```
I suppose the goal is to display the related form view.
You can use:
`return self.subscription_id.get_formview_action()`

acsonefho commented on subscription_oca/models/sale_subscription.py:
```diff
@@ -134,24 +134,27 @@ def _read_group_stage_ids(self, stages, domain, order):
         today = date.today()
-        for subscription in self.search([]):
+        for subscription in self.search([], order="recurring_next_date asc"):
+            subscription = subscription.with_company(subscription.company_id)
```
Good catch :muscle:

acsonefho commented on subscription_oca/models/sale_subscription.py:
```diff
@@ -244,6 +246,15 @@ def action_close_subscription(self):
+        closed_stage = self.env["sale.subscription.stage"].search(
+            [("type", "=", "post")], limit=1
+        )
+        self.close_reason_id = close_reason_id
```
```
self.write({
    "close_reason_id": close_reason_id,
    "stage_id": closed_stage.id,
})
```

acsonefho commented:
Thank you @tarteo for your contribution :muscle: Just minor changes :pray:

tobiaszehntner commented:
@tarteo we have to abandon our migration PR because we no longer need it and don't have the time. Feel free to apply the requested changes and create your own PR to 18, I think we're pretty close.

tarteo commented:
Alright, well nonetheless thanks for your review and time! I'll apply your suggestions when I'll create a new PR. Have a nice weekend!

pere-aquarian committed:
Added translation using Weblate (Catalan)

Commit:
[UPD] Update contract_queue_job.pot

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] agreement_rebate_partner_company_group: pre-commit auto fixes

victoralmau committed:
[MIG] agreement_rebate_partner_company_group: Migration to 18.0

TT54413

MiquelRForgeFlow committed:
[OU-FIX] contract: migration scripts

OCA-git-bot committed:
Merge PR #1247 into 18.0

Signed-off-by yvaucher

Commit:
[UPD] Update contract_invoice_start_end_dates.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1273 into 18.0

Signed-off-by yvaucher

OCA-git-bot committed:
Merge PR #1140 into 18.0

Signed-off-by yvaucher

Commit:
[UPD] Update contract_variable_qty_prorated.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1145 into 18.0

Signed-off-by yvaucher

Commit:
[UPD] Update product_contract_variable_quantity.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

pedrobaeza committed:
Merge pull request #1278 from ForgeFlow/18.0-fix-contract-migration

[18.0][FIX] contract: migration scripts

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: contract-18.0/contract-18.0-contract_variable_qty_prorated
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract_variable_qty_prorated/it/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 2.7% (1 of 37 strings)

Translation: contract-18.0/contract-18.0-contract_forecast
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract_forecast/sv/

qgroulard committed:
[IMP] Contract line: compute display name

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 98.4% (251 of 255 strings)

Translation: contract-18.0/contract-18.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (37 of 37 strings)

Translation: contract-18.0/contract-18.0-contract_forecast
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract_forecast/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (1 of 1 strings)

Translation: contract-18.0/contract-18.0-contract_forecast_variable_quantity
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract_forecast_variable_quantity/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (4 of 4 strings)

Translation: contract-18.0/contract-18.0-contract_invoice_auto_validate
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract_invoice_auto_validate/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (87 of 87 strings)

Translation: contract-18.0/contract-18.0-contract_line_successor
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract_line_successor/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (2 of 2 strings)

Translation: contract-18.0/contract-18.0-contract_payment_mode
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract_payment_mode/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (29 of 29 strings)

Translation: contract-18.0/contract-18.0-contract_termination
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract_termination/sv/

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (112 of 112 strings)

Translation: contract-18.0/contract-18.0-product_contract
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-product_contract/sv/

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Added translation using Weblate (Swedish)

glitchov committed:
[FIX] contract_forecast: multi-company rule for all authorized companies

bjouini-acsone committed:
[IMP] product_contract:seperate function for line name formatting

bo-eska committed:
[FIX] product_contract: divide contract configuration button

OCA-git-bot committed:
Merge PR #1281 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update product_contract.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-18.0/contract-18.0-product_contract
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-product_contract/

OCA-git-bot committed:
Merge PR #1283 into 18.0

Signed-off-by yvaucher

Commit:
[UPD] Update product_contract.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-18.0/contract-18.0-product_contract
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-product_contract/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (113 of 113 strings)

Translation: contract-18.0/contract-18.0-product_contract
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-product_contract/it/

OCA-git-bot committed:
Merge PR #1206 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update contract_sale_invoicing.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1207 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update contract.pot

Commit:
[UPD] Update contract_variable_qty_timesheet.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-18.0/contract-18.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-18-0/contract-18-0-contract/

qgroulard committed:
[FIX] product_contract_variable_quantity: product contract configurator

Following the breaking change (method renamed) introduced in 52cce0a28779498d400e5da3371d9fdba18058bf.

OCA-git-bot committed:
Merge PR #1280 into 18.0

Signed-off-by yvaucher

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1292 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1290 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update contract_price_revision.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1269 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update agreement_rebate_partner_company_group.pot

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] contract_analytic_tag: pre-commit auto fixes

victoralmau committed:
[MIG] contract_analytic_tag: Migration to 18.0

TT57828

OCA-git-bot committed:
Merge PR #1295 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update contract_analytic_tag.pot

OCA-git-bot committed:
[BOT] post-merge updates

tobiaszehntner committed:
[FIX] contract.recurring_create_invoice: Add ensure_one

OCA-git-bot committed:
Merge PR #1236 into 18.0

Signed-off-by yvaucher

Commit:
[UPD] Update contract_transmit_method.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1298 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1282 into 18.0

Signed-off-by yvaucher

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1228 into 18.0

Signed-off-by yvaucher

Commit:
[UPD] Update contract_mandate.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1227 into 18.0

Signed-off-by sbejaoui

Commit:
[UPD] Update contract_queue_job.pot

OCA-git-bot committed:
[BOT] post-merge updates

tobiaszehntner committed:
18 contract multi invoice line wip
